testsuite/MDAnalysisTests/analysis/test_align.py

```diff
@@ -500,11 +500,31 @@ def test_fasta2select_ClustalW(self, tmpdir):
         # numbers computed from alignment with clustalw 2.1 on Mac OS X
         # [orbeckst] length of the output strings, not residues or anything
         # real...
-        assert len(sel['reference']) == 23080, ("selection string has"
-                                                "unexpected length")
-        assert len(
-            sel['mobile']) == 23090, "selection string has unexpected length"
-
+        assert len(sel['reference']) == 23080, self.error_msg
+        assert len(sel['mobile']) == 23090, self.error_msg
+
+    def test_fasta2select_nogap(self, tmpdir):
+        """test align.fasta2select() on aligned FASTA with no gap in residue
+        (Issue #3124)"""
+        reference = mda.Universe(self.ref).select_atoms('all')
+        target = mda.Universe(self.tgt).select_atoms('all')
+        CA_ref = reference.select_atoms('protein and name CA and not altloc B')
+        CA_target = target.select_atoms('protein and name CA and not altloc B')
+        ref_resids = [a.resid for a in CA_ref]
+        target_resids = [a.resid for a in CA_target]
+        alignment = align.sequence_alignment(
+            target.select_atoms('protein and not altloc B'),
+            reference.select_atoms('protein and not altloc B'))
+        with tmpdir.as_cwd():
+            fasta = f">ref\n{alignment[0]}\n>mobile\n{alignment[1]}"
+            with open('alignment.aln', "w+") as f:
+                f.write(fasta)
+            sel = align.fasta2select('alignment.aln',
```

Comment:
To clarify my comment from the old PR, with this test you're loading two files of the same sequence, aligning then to create a temporary 'alignment.aln' file without any gaps, and then using that as the test input for fasta2select - so alignment.aln gets recreated from scratch every time. If you include alignment.aln as a test datafile, rather than the structure files, you can use that directly and avoid needing all the setup steps.

Reply:
@fiona-naughton the error occurs due to no gaps in the resids which is extracted from the two files. So to extract the resids we'll have to load the structure files right?

Reply:
Apologies for the delay getting back to you. So having had a closer look it seems like the initial issue was actually when `target_resids` and `ref_resids` are both defined - it's these values the `if` statement ends up checking for `None`, rather than looking for gaps. Your fix still works, though.

For the test: so you do indeed need to pass in resids, but this is just a list of residue indices. Since we're just checking that the function runs, it doesn't matter what the exact values are, so you could just generate any list of the right length rather than extracting them from the structure files.

Reply:
@fiona-naughton , so I removed the unnecessary files and just used the `test.fasta` file used in other tests with a random list for resids. Is this fine now?